doc/storage_plugins.md

```diff
@@ -4,20 +4,183 @@ Storage plugins
 Content
 -------
   1. [Introduction](#introduction)
-  2. [Using Python](#using-python)
-  3. [Writing in C](#writing-in-c)
+  2. [Working with storages in Python](#working-with-storages-in-python)
+  3. [Using storages implicitely](#using-storages-implicitely)
+  4. [Writing Python storage plugins](#writing-python-storage-plugins)
+  5. [Working with storages from C and Fortran](#working-with-storages-from-c-and-fortran)
 
 
 Introduction
 ------------
-xxx
+A storage is in DLite an abstraction for an external data source or sink.
+It can be a file on disk, a local database or a database accessed via a web interface.
+Loading data from a storage into an instance and saving it back again is a key mechanism for interoperability at a syntactic level.
 
+DLite provide a plugin system that make it easy to connect to new data sources via a common interface (using a [strategy design pattern]).
+Opening a storage takes three arguments, a `driver` name identifying the storage plugin to use, the `location` of the storage and `options`.
 
-Using Python
-------------
-xxx
+Storage plugins can be categorised as either *generic* or *specific*.
+A generic storage plugin is can store and retrieve any type of instances and metadata while specific storage plugin is typically specific instances of one type of entity.
+DLite comes with a set of generic storage plugins, like json, yaml, rdf, hdf5, postgresql and mongodb.
+It also comes with a specific Blob storage plugin, that can load and save instances of a `http://onto-ns.com/meta/0.1/Blob` entity.
+Storage plugins can be written in either C or Python.
 
 
-Writing in C
-------------
-xxx
+Working with storages in Python
+-------------------------------
+For instance, to create a new file-based JSON storage can in Python be done with:
+
+```python
+    >>> import dlite
+    >>> s = dlite.Storage("json", location="newfile.json", options="mode=w")
+
+```
+
+where the first argument is the name of the driver to use.
+It is also possible to combine the three arguments into an URL as in
+
+```python
+    >>> s = dlite.Storage("json://newfile.json?mode=w")
+
+```
+
+Note that the URL form less flexible, since the scheme part identifies the driver and can therefore not be used in the location of an external storage.
```

Comment:
```suggestion
Note that the URL form is less flexible, since the scheme part identifies the driver and can therefore not be used in the location of an external storage.
```